src/wp-includes/update.php

```diff
@@ -1074,12 +1074,78 @@ function _wp_delete_all_temp_backups() {
 	}
 }
 
+/**
+ * Checks for translation file updates and adds to update transient
+ *
+ * Hooks to `wp_version_check` and looks for any installed translation file locales,
+ * compares the file PO-Revision-Date to the API date. Any translation file updates
+ * are injected into the `update_core` transient` for later processing.
+ *
+ * @since CP-2.5.0
+ */
+function cp_translation_file_updates() {
+	require_once ABSPATH . 'wp-admin/includes/translation-install.php';
+	$installed_translations = wp_get_installed_translations( 'core' );
+
+	// return if no translation files installed
+	if ( empty( $installed_translations ) ) {
+		return;
+	}
+
+	$domains = array( 'admin', 'admin-network', 'continents-cities', 'default' );
+	$po_data = array();
+
+	foreach ( $domains as $domain ) {
+		foreach ( $installed_translations[ $domain ] as $locale => $data ) {
+			if ( ! isset( $po_data[ $locale ] ) ) {
+				$po_data[ $locale ] = strtotime( $data['PO-Revision-Date'] );
+			} else {
+				if ( strtotime( $data['PO-Revision-Date'] ) > $po_data[ $locale ] ) {
+					$po_data[ $locale ] = strtotime( $data['PO-Revision-Date'] );
+				}
+			}
+		}
+	}
+
+	$translations = wp_get_available_translations();
+
+	$translation_updates = array();
+
+	foreach ( $translations as $translation ) {
+		if ( array_key_exists( $translation['language'], $po_data ) ) {
+			if ( strtotime( $translation['updated'] ) > $po_data[ $translation['language'] ] ) {
+				$translation_updates[] = array(
+					'type'       => 'core',
+					'slug'       => 'default',
+					'language'   => $translation['language'],
+					'package'    => $translation['package'],
+					'autoupdate' => true,
+				);
+			}
+		}
+	}
+
+	// return if no translation updates available
+	if ( empty( $translation_updates ) ) {
+		return;
+	}
+
+	$updates = get_site_transient( 'update_core' );
+	if ( isset( $updates->translations ) ) {
+		$updates->translations = array_merge( $updates->translations, $translation_updates );
```

Comment:
Please try:
`$updates->translations = array_unique( array_merge( $updates->translations, $translation_updates ), SORT_REGULAR );`

`SORT_REGULAR` fixes an "array to string conversion" notice.

Reply:
Agree with this also, tested overnight in commit to follow.